### DISCUSSION TITLE
function inside type='module' as x-data

### DISCUSSION THREAD

**Discussion by sqllyw** [alpinejs/alpine#1126] [1. Help] [answered]
Hi,

I got a very simple app here:

https://jsfiddle.net/ac56mxjh/1/

if x-data="dropdown()" it works, if x-data="dropdown0()" it can not find dropdown0, any idea ?thanks

**Comment by SimoTod** [comment, marked as answer]
I don't think that's a valid module. You need to export the function you want to expose and import it when you want to use it. Also, modules are deferred by default so your code if it ran, it would do after alpine.

**Reply by KevinBatdorf** [reply]
> Also, modules are deferred by default so your code if it ran, it would do after alpine.

☝ if you move `dropdown0` above Alpine it works.

**Reply by sqllyw** [reply]
> > Also, modules are deferred by default so your code if it ran, it would do after alpine.
> 
>  if you move `dropdown0` above Alpine it works.

this works,  purpose to put x-data function inside a type='module' is to factor out js code into pure js files and import it to the x-data function, here is a updated version of my original question:

https://codepen.io/sqllyw/pen/rNWYZyd

Thanks
